conans/client/cache/cache.py

```diff
@@ -134,6 +133,22 @@ def get_latest_prev(self, ref):
         prevs = self.get_package_revisions(ref, True)
         return prevs[0] if prevs else None
 
+    def get_latest_pkg_layout(self, ref: ConanReference) -> PackageLayout:
```

Comment:
This looks mostly like test code in production scope:

- Better move them to ``TestClient``
- When necessary, in production code, call explicitly ``cache.get_latest_rrev()`` to get the latest revision (in the ``conan get``)

This way we will have more control over who is resolving to latest revision in the cache, at the moment it is only used in a couple of places (uploader, proxy).

Reply:
Yeah, you're completely right. I should have realized about having something like this one:

```python
    def get_latest_pkg_layout(self, ref: PackageReference) -> PackageLayout:
        """Get the latest PackageLayout given a package reference"""
        latest_prev = self.get_latest_prev(ref)
        pkg_layout = self.get_pkg_layout(latest_prev)
        return pkg_layout
```

But yes, agree with having it explicit in production scope 👍